central/graphql/resolvers/image_scan_benchmark_test.go

```diff
@@ -0,0 +1,176 @@
+package resolvers
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/testutils/envisolator"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	imageOnlyQuery = `
+		query getImages($query: String, $pagination: Pagination) {
+			images(query: $query, pagination: $pagination) { 
+				id
+			}}`
+
+	imageWithCountsQuery = `
+		query getImages($query: String, $pagination: Pagination) {
+			images(query: $query, pagination: $pagination) { 
+				id
+				imageComponentCount
+				imageVulnerabilityCount
+			}}`
+
+	imageWithScanLongQuery = `
+		query getImages($query: String, $pagination: Pagination) {
+			images(query: $query, pagination: $pagination) { 
+				id
+				scan {
+					imageComponents {
+						name
+						lastScanned
+						imageVulnerabilities {
+							cve
+							fixedByVersion
+						}
+					}
+				}
+			}}`
+
+	imageWithoutScanLongQuery = `
+		query getImages($query: String, $pagination: Pagination) {
+			images(query: $query, pagination: $pagination) { 
+				id
+				imageComponents {
+					name
+					lastScanned
+					imageVulnerabilities {
+						cve
+						fixedByVersion
+					}
+				}
+			}}`
+)
+
+func BenchmarkImageResolver(b *testing.B) {
+	envIsolator := envisolator.NewEnvIsolator(b)
+	envIsolator.Setenv(features.PostgresDatastore.EnvVar(), "true")
+	defer envIsolator.RestoreAll()
+
+	if !features.PostgresDatastore.Enabled() {
+		b.Skip("Skip postgres store tests")
+		b.SkipNow()
+	}
+
+	mockCtrl := gomock.NewController(b)
+	defer mockCtrl.Finish()
+	db, gormDB := setupPostgresConn(b)
+	defer pgtest.CloseGormDB(b, gormDB)
+	defer db.Close()
+
+	imageDataStore := createImageDatastore(b, mockCtrl, db, gormDB)
```

Comment:
any particular reason for all these stores we mock out risk? Seems a bit odd to be mixing mocked objects with real objects

Reply:
We are not testing whether risk datastore. To enable testing risk datastore it's store needs to be moved out of internal which seems unnecessary for what this test's focus is.